src/components/Contacts/ContactDetails/ContactDetailsHeader/DeleteContactModal/DeleteContactModal.tsx

```diff
@@ -22,15 +25,71 @@ interface DeleteContactModalProps {
   setOpen: (open: boolean) => void;
   deleting: boolean;
   deleteContact: () => void;
+  contactId: string;
 }
 
 export const DeleteContactModal: React.FC<DeleteContactModalProps> = ({
   open,
   setOpen,
   deleting,
   deleteContact,
+  contactId,
 }) => {
   const { t } = useTranslation();
+  const accountListId = useAccountListId() ?? '';
+  const [canDelete, setCanDelete] = useState(true);
+
+  const { data } = useContactSourceQuery({
+    variables: { accountListId, contactId },
+    skip: !open && !contactId,
+  });
+  const contactSources = data?.contact;
+
+  useEffect(() => {
```

Comment:
Instead of having an effect that calculates the value of `canDelete` and sets it, this should be a `useMemo` that returns `canDelete`. Changing that will avoid an extra render.